src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceData/CounterSet.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Microsoft.Win32;
+
+namespace System.Diagnostics.PerformanceData
+{
+    /// <summary>
+    /// CounterSet is equivalent to "Counter Object" in native performance counter terminology,
+    /// or "Counter Category" in previous framework releases. It defines a abstract grouping of
+    /// counters, where each counter defines measurable matrix. In the new performance counter
+    /// infrastructure, CounterSet is defined by GUID called CounterSetGuid, and is hosted inside
+    /// provider application, which is also defined by another GUID called ProviderGuid.
+    /// </summary>    
+    public class CounterSet : IDisposable
+    {
+        private static readonly bool s_platformNotSupported = (Environment.OSVersion.Version.Major < 6);
+        internal PerfProvider m_provider;
+        internal Guid m_providerGuid;
+        internal Guid m_counterSet;
+        internal CounterSetInstanceType m_instType;
+        private readonly object m_lockObject;
+        private bool m_instanceCreated;
+        internal Dictionary<string, int> m_stringToId;
+        internal Dictionary<int, CounterType> m_idToCounter;
+
+        /// <summary>
+        /// CounterSet constructor.
+        /// </summary>
+        /// <param name="providerGuid">ProviderGuid identifies the provider application. A provider identified by ProviderGuid could publish several CounterSets defined by different CounterSetGuids</param>
+        /// <param name="counterSetGuid">CounterSetGuid identifies the specific CounterSet. CounterSetGuid should be unique.</param>
+        /// <param name="instanceType">One of defined CounterSetInstanceType values</param>        
+        [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "guid", Justification = "Approved")]
```

Comment:
@danmosemsft can we remove this ?

Reply:
`[System.Security.SecuritySafeCritical]` yes..i'm missed that...i don't know about `[SuppressMessage]`

Reply:
removed [System.Security.SecuritySafeCritical]

Reply:
You can remove all `SuppressMessage` as well I think. They refer to old fxcop rules that we do not run in corefx.